contracts/starknet/strategies/single_slot_proof.cairo

```diff
@@ -0,0 +1,92 @@
+%lang starknet
+%builtins pedersen range_check bitwise
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256, uint256_add
+from starkware.cairo.common.math import unsigned_div_rem, assert_nn_le
+from starkware.cairo.common.pow import pow
+
+from contracts.starknet.fossil.contracts.starknet.types import StorageSlot
+from contracts.starknet.lib.type_conversions import ints_to_uint256
+
+# FactRegistry simplified interface
+@contract_interface
+namespace IFactsRegistry:
+    func get_storage(
+            block : felt, account_160 : felt, slot : StorageSlot, proof_sizes_bytes_len : felt,
+            proof_sizes_bytes : felt*, proof_sizes_words_len : felt, proof_sizes_words : felt*,
+            proofs_concat_len : felt, proofs_concat : felt*) -> (
+            res_bytes_len : felt, res_len : felt, res : felt*):
+    end
+end
+
+# Address of the fact registry. This is an immutable value that can be set at contract deployment only.
```

Comment:
In a future version, should this be updatable (in case of a re-deployment of the fossil contract?

Reply:
that sounds like a bad idea to me, this is a library contract after all and will have no 'owner' so who would be have the permissions to change it? In the case of fossil being redeployed we would then redeploy the strategies that depend on it.